project/ParadoxSupport.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka
+
+import _root_.io.github.lukehutch.fastclasspathscanner.FastClasspathScanner
+import com.lightbend.paradox.markdown._
+import com.lightbend.paradox.sbt.ParadoxPlugin.autoImport._
+import org.pegdown.Printer
+import org.pegdown.ast._
+import sbt.Keys._
+import sbt._
+
+import scala.collection.JavaConverters._
+
+object ParadoxSupport {
+  val paradoxWithCustomDirectives = Seq(
+    paradoxDirectives ++= Def.taskDyn {
+      val classpath = (fullClasspath in Compile).value.files.map(_.toURI.toURL).toArray
+      val classloader = new java.net.URLClassLoader(classpath, this.getClass().getClassLoader())
+      lazy val scanner = new FastClasspathScanner("akka").addClassLoader(classloader).scan()
+      val allClasses = scanner.getNamesOfAllClasses.asScala.toVector
+      Def.task { Seq(
+        { _: Writer.Context ⇒ new UnidocDirective(allClasses) },
+        { context: Writer.Context ⇒ new UnidocRefDirective(context.location.tree.label, context.properties) }
+      )}
+    }.value
+  )
+
+  class UnidocDirective(allClasses: IndexedSeq[String]) extends InlineDirective("unidoc") {
```

Comment:
This is copied from akka/akka-http#1579 removing unnecessary stuff.

Reply:
What unnecessary stuff could be removed? can we also remove it over there?

Reply:
Unnecessary stuff was removed in   https://github.com/akka/akka/pull/24563/commits/41611755fd016f6374542036981487759a211e90 - unused parameters, and `case 2` in the pattern match could be handled by `case n`.

As we are making the new sbt plugin anyway, we don't need to do the same in akka-http? However, if the plugin takes too much time, yes I'll do that in akka-http.